transform/transform.go

```diff
@@ -12,74 +12,83 @@ ReverseComplement takes the reverse complement of a sequence.
 */
 package transform
 
-import "strings"
-
-// complementBaseRuneMap provides 1:1 mapping between bases and their complements
-var complementBaseRuneMap = map[rune]rune{
-	65:  84,  // A -> T
-	66:  86,  // B -> V
-	67:  71,  // C -> G
-	68:  72,  // D -> H
-	71:  67,  // G -> C
-	72:  68,  // H -> D
-	75:  77,  // K -> M
-	77:  75,  // M -> K
-	78:  78,  // N -> N
-	82:  89,  // R -> Y
-	83:  83,  // S -> S
-	84:  65,  // T -> A
-	85:  65,  // U -> A
-	86:  66,  // V -> B
-	87:  87,  // W -> W
-	89:  82,  // Y -> R
-	97:  116, // a -> t
-	98:  118, // b -> v
-	99:  103, // c -> g
-	100: 104, // d -> h
-	103: 99,  // g -> a
-	104: 100, // h -> d
-	107: 109, // k -> m
-	109: 107, // m -> k
-	110: 110, // n -> n
-	114: 121, // r -> y
-	115: 115, // s -> s
-	116: 97,  // t -> a
-	117: 97,  // u -> a
-	118: 98,  // v -> b
-	119: 119, // w -> w
-	121: 114, // y -> r
-}
+import "unsafe"
 
 // ReverseComplement takes the reverse complement of a sequence.
+// ReverseComplement expects ASCII input.
 func ReverseComplement(sequence string) string {
-	complementString := strings.Map(ComplementBase, sequence)
-	length := len(complementString)
-	newString := make([]rune, length)
-	for _, base := range complementString {
-		length--
-		newString[length] = base
+	n := len(sequence)
+	newSeq := make([]byte, n)
+	for i := 0; i < n; i++ {
```

Comment:
@TimothyStiles doesn't let any single letter variables in. Could you change the i and n to something longer / more descriptive? (like above)